### ISSUE TITLE
The Provider.forUseAtConfigurationTime method and the org.gradle.util.NameMatcher type has been deprecated. This is scheduled to be removed in Gradle 9.0

### ISSUE THREAD

**Issue by denzap** [cdsap/Talaiot#392]
Gradle: 8.6
AGP: 8.2.2
Talaiot: 2.0.3

After running project build with `--scan` option (or with `./gradlew --warning-mode all --stacktrace`) we got 3 deprecations from gradle: 

```
The Provider.forUseAtConfigurationTime method has been deprecated.	
This is scheduled to be removed in Gradle 9.0.	
Simply remove the call.	
[Documentation](https://docs.gradle.org/8.6/userguide/upgrading_version_7.html#for_use_at_configuration_time_deprecation)	
2 usages	
at org.gradle.api.internal.provider.AbstractMinimalProvider.forUseAtConfigurationTime(AbstractMinimalProvider.java:135)	
at io.github.cdsap.talaiot.metrics.UserMetric$1.invoke(GradleMetrics.kt:126)	
at io.github.cdsap.talaiot.metrics.UserMetric$1.invoke(GradleMetrics.kt:126)	
at io.github.cdsap.talaiot.metrics.base.Metric.get(Metric.kt:15)
...
```

```
The Provider.forUseAtConfigurationTime method has been deprecated.	
This is scheduled to be removed in Gradle 9.0.	
Simply remove the call.	
[Documentation](https://docs.gradle.org/8.6/userguide/upgrading_version_7.html#for_use_at_configuration_time_deprecation)	
2 usages
at org.gradle.api.internal.provider.AbstractMinimalProvider.forUseAtConfigurationTime(AbstractMinimalProvider.java:135)	
at io.github.cdsap.talaiot.metrics.LocaleMetric$1.invoke(GradleMetrics.kt:132)	
at io.github.cdsap.talaiot.metrics.LocaleMetric$1.invoke(GradleMetrics.kt:131)
...
```

```
The org.gradle.util.NameMatcher type has been deprecated.	
This is scheduled to be removed in Gradle 9.0.	
[Documentation](https://docs.gradle.org/8.6/userguide/upgrading_version_7.html#org_gradle_util_reports_deprecations)	
1 usage	
at org.gradle.util.NameMatcher.<clinit>(NameMatcher.java:38)	
at io.github.cdsap.talaiot.util.TaskAbbreviationMatcher.<init>(TaskAbbreviationMatcher.kt:10)	
at io.github.cdsap.talaiot.metrics.GradleMetricsKt.findRequestedTasks(GradleMetrics.kt:140)
...
```

**Expected behavior**
No warnings

**Comment by cdsap** [comment]
Thanks for the report will take a look soon

**Comment by cdsap** [comment]
added in #395, it will be included in the next release

**Comment by TWiStErRob** [comment]
@cdsap can you please release this? We would like to use `--warning-mode=fail` in Gradle, but this issue is blocking us from doing that.

**Comment by cdsap** [comment]
@TWiStErRob just released 2.0.5 including this change

**Comment by TWiStErRob** [comment]
Thank you so much! I think this issue can be closed then?
